pam-u2f.c

```diff
@@ -278,6 +282,38 @@ int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
     }
   }
 
+  // Determine the lock file path for touch request notifications
+  if (!cfg->lock_file) {
+    buf = NULL;
+
+    char *run_folder = "/var/run/user/";
+    uid_t uid = getuid();
+    char *lock_file_name = "/pam-u2f-touch";
```

Comment:
I'd rather use the variable `buffer` and add a `#define DEFAULT_TOUCH_FILE_PATH "/var/run/user/%d/pam-u2f-touch"` to `snprintf` into it and `strdup` it